package/CHANGELOG

```diff
@@ -26,6 +26,8 @@ Fixes
     array warnings (PR #3139, backports PRs #2845 and #2834)
   * Fixed several issues with NSGrid and triclinic boxes not finding some pairs.
     (Issues #2229 #2345 #2919, PR #2937).
+  * Removed deprecation warning from numpy in hbond_autocorrel
```

Comment:
Given the changes introduced here, did the previous exclusions code give the right behaviour? If not, we should warn users of behaviour changes here.

Reply:
I think (@richardjgowers please chime in) that -- the old and new code has different behaviour, *but*, it "shouldn't" result in users noticing different results. In the new `_in2d` function, the ordering of the H-donor and H-acceptor indices matters when matching exclusions. In the old one it doesn't. So, in this new code, it matters if users mess up which atom is the donor and which is the acceptor when inputting `exclusions`.

Reply:
@IAlibay did you mean warn users with a real warning or in the CHANGELOG?

Reply:
So originally this used to work on a dense matrix, so the input of two tuples of "coordinates" into the dense matrix could be applied directly.  

When we switched to a sparse matrix (i.e. capped_dist) I think we started doing it wrong because `pair == exclude` isn't masking out pairs as expected.  So you might get excluded pairs in the output.

This new code uses a "set of tuples" to exclude pairs so will work.

Reply:
> @IAlibay did you mean warn users with a real warning or in the CHANGELOG?

Sorry I meant just a changelog entry. I know most don't read it, but it is a useful record for folks to keep track of stuff that might have affected their work.